label_studio/io_storages/azure_blob/utils.py

```diff
@@ -42,3 +45,36 @@ def get_blob_metadata(cls, url: str, container: str, account_name: str = None, a
         _, container = cls.get_client_and_container(container, account_name=account_name, account_key=account_key)
         blob = container.get_blob_client(url)
         return dict(blob.get_blob_properties())
+
+    @classmethod
+    def validate_pattern(cls, storage, pattern, glob_pattern=True):
+        """
+        Validate pattern against Azure Blob Storage
+        :param storage: AzureBlobStorage instance
+        :param pattern: Pattern to validate
+        :param glob_pattern: If True, pattern is a glob pattern, otherwise it is a regex pattern
+        :return: Message if pattern is not valid, empty string otherwise
+        """
+        logger.debug('Validating Azure Blob Storage pattern.')
+        client, container = storage.get_client_and_container()
+        if storage.prefix:
+            generator = container.list_blob_names(name_starts_with=storage.prefix,
+                                                  results_per_page=settings.CLOUD_PAGE_CHECKED_OBJECTS,
+                                                  timeout=settings.CLOUD_TIMEOUT_CHECKED_OBJECTS)
+        else:
+            generator = container.list_blob_names(results_per_page=settings.CLOUD_PAGE_CHECKED_OBJECTS,
+                                                  timeout=settings.CLOUD_TIMEOUT_CHECKED_OBJECTS)
+        # compile pattern to regex
+        if glob_pattern:
+            pattern = fnmatch.translate(pattern)
+        regex = re.compile(str(pattern))
+        # match pattern against all keys in the container
+        for index, key in enumerate(generator):
+            # skip directories
+            if key.endswith('/'):
+                logger.debug(key + ' is skipped because it is a folder')
+                continue
+            if regex and regex.match(key):
+                logger.debug(key + ' matches file pattern')
+                return ""
```

Comment:
On one hand returning a string keeps thing simple, which is great, but on the other it seems weird to signal "yes, this is valid" using an empty string.
On the other other hand, I guess this just gets used by django under the hood; is this a pretty standard django pattern?

Reply:
It allows us to keep the API generic while having the ability to return a custom error message including some params from the function itself